File: samplestore/src/main/java/com/stripe/samplestore/ProgressDialogFragment.java
package com.stripe.samplestore;

import android.app.Dialog;
import android.app.ProgressDialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;

public class ProgressDialogFragment extends DialogFragment {

    @NonNull
    public static ProgressDialogFragment newInstance(@NonNull String message) {
        ProgressDialogFragment fragment = new ProgressDialogFragment();

        final Bundle args = new Bundle();
        args.putString("message", message);

        fragment.setArguments(args);

        return fragment;
    }

    public ProgressDialogFragment() {
        // Empty constructor required for DialogFragment
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        final ProgressDialog dialog = new ProgressDialog(getActivity());
        dialog.setMessage(getMessage());
        return dialog;
    }

    @Nullable
    private String getMessage() {
        return getArguments() != null ? getArguments().getString("message") : null;
    }
}

File: example/src/main/java/com/stripe/example/dialog/ProgressDialogFragment.kt
package com.stripe.example.dialog

import android.app.Dialog
import android.app.ProgressDialog
import android.os.Bundle

class ProgressDialogFragment : androidx.fragment.app.DialogFragment() {

    private val message: String?
        get() = arguments?.getString("message")

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialog = ProgressDialog(activity)
        dialog.setMessage(message)
        return dialog
    }

    companion object {
        fun newInstance(message: String): ProgressDialogFragment {
            val fragment = ProgressDialogFragment()

            val args = Bundle()
            args.putString("message", message)

            fragment.arguments = args

            return fragment
        }
    }
}
